File: lifecycledemo/src/main/java/com/jarvis/lifecycledemo/MainActivity.java
package com.jarvis.lifecycledemo;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;


public class MainActivity extends AppCompatActivity {

    private MyLocationListener myLocationListener;
    private TextView tvHello;

    private static void onLocateSuccess(String location) {
        //update UI
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
//
        tvHello = findViewById(R.id.tv_hello);
//        myLocationListener = new MyLocationListener(this, location -> {
//            //update UI
//        });

        getLifecycle().addObserver(new MyLocationListener2(this, MainActivity::onLocateSuccess));
    }

    @Override
    protected void onStart() {
        super.onStart();
//        myLocationListener.start();
//        tvHello.postDelayed(() -> myLocationListener.start(), 1000 * 4);
//        tvHello.postDelayed(MainActivity.this::finish, 1000 * 2);
    }

    @Override
    protected void onStop() {
        super.onStop();
//        myLocationListener.stop();
    }
}

File: databindingdemo/src/main/java/com/jarvis/databindingdemo/MainActivity.kt
package com.jarvis.databindingdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.jarvis.databindingdemo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {


    lateinit var user: User
    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
        user = User("1", "11")
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.user = user
//        user1.notifyPropertyChanged(BR.user)


        Thread {
            for (index in 0..10) {
                Thread.sleep(1000)
//                binding.user = User("$index","$index")
                user.name = "Jarvis$index"
                binding.user1 = User1("user1","user1")
//                user1.name = (DataBindingUtil.getDefaultComponent() == null).toString()
            }
        }.start()
    }
}